benches/benches_badger_rocks.rs

```diff
@@ -0,0 +1,255 @@
+        assert_eq!(item.value(), value);
+    }
+}
+
+fn badger_iterate(agate: Arc<Agate>, value_size: usize) {
+    let txn = agate.new_transaction_at(unix_time(), false);
+    let opts = IteratorOptions::default();
+    let mut iter = txn.new_iterator(&opts);
+    iter.rewind();
+
+    while iter.valid() {
+        let item = iter.item();
+        assert_eq!(item.value().len(), value_size);
+
+        iter.next();
+    }
+}
+
+fn rocks_populate(db: Arc<DB>, value_size: usize) {
+    let mut write_options = rocksdb::WriteOptions::default();
+    write_options.set_sync(true);
+    write_options.disable_wal(false);
+
+    let mut batch = rocksdb::WriteBatch::default();
+
+    for i in 0..BATCH_SIZE {
+        let (key, value) = gen_kv_pair(i, value_size);
+        batch.put(key, value);
+    }
+
+    db.write_opt(batch, &write_options).unwrap();
+}
+
+fn rocks_randread(db: Arc<DB>, value_size: usize, rng: &mut ThreadRng) {
+    for _ in 0..BATCH_SIZE {
+        let (key, value) = gen_kv_pair(rng.gen_range(0, BATCH_SIZE), value_size);
+
+        let find = db.get(key).unwrap();
+        assert_eq!(find.unwrap(), value)
+    }
+}
+
+fn rocks_iterate(db: Arc<DB>, value_size: usize) {
+    let iter = db.iterator(rocksdb::IteratorMode::Start);
+
+    for (_, value) in iter {
+        assert_eq!(value.len(), value_size);
+    }
+}
+
+fn bench_badger(c: &mut Criterion) {
+    let mut rng = rand::thread_rng();
+
+    let dir = TempDir::new("agatedb-bench-small-value").unwrap();
+    let dir_path = dir.path();
+    let mut opts = AgateOptions {
+        create_if_not_exists: true,
+        sync_writes: true,
+        dir: dir_path.to_path_buf(),
+        value_dir: dir_path.to_path_buf(),
+        managed_txns: true,
+        ..Default::default()
+    };
+
+    c.bench_function("badger populate small value", |b| {
+        b.iter_custom(|iters| {
+            let mut total = Duration::new(0, 0);
+
+            (0..iters).into_iter().for_each(|_| {
+                remove_files(dir_path);
+                let agate = Arc::new(opts.open().unwrap());
+
+                let now = Instant::now();
+                badger_populate(agate, SMALL_VALUE_SIZE);
+                total = total.add(now.elapsed());
+            });
+
+            total
+        });
+    });
+
+    let agate = Arc::new(opts.open().unwrap());
+
+    c.bench_function("badger randread small value", |b| {
+        b.iter(|| {
+            badger_randread(agate.clone(), SMALL_VALUE_SIZE, &mut rng);
+        });
+    });
+
+    c.bench_function("badger iterate small value", |b| {
+        b.iter(|| {
+            badger_iterate(agate.clone(), SMALL_VALUE_SIZE);
+        });
+    });
+
+    dir.close().unwrap();
+    let dir = TempDir::new("agatedb-bench-large-value").unwrap();
+    let dir_path = dir.path();
+    opts.dir = dir_path.to_path_buf();
+    opts.value_dir = dir_path.to_path_buf();
+
+    c.bench_function("badger populate large value", |b| {
```

Comment:
should reuse the `bench_badger` impl with value size as a variable.

Reply:
In fact, I found we could not simply reuse them.